server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -47,172 +45,208 @@
     private final MessageFactory messageFactory;
+    private final SystemMailboxesProvider systemMailboxesProvider;
 
-    @Inject
-    @VisibleForTesting
-    SetMessagesCreationProcessor(MailboxMapperFactory mailboxMapperFactory,
-                                 MailboxManager mailboxManager,
-                                 MailboxSessionMapperFactory mailboxSessionMapperFactory,
+    
+    @VisibleForTesting @Inject
+    SetMessagesCreationProcessor(MailboxSessionMapperFactory mailboxSessionMapperFactory,
                                  MIMEMessageConverter mimeMessageConverter,
                                  MailSpool mailSpool,
                                  MailFactory mailFactory,
-                                 MessageFactory messageFactory) {
-        this.mailboxMapperFactory = mailboxMapperFactory;
-        this.mailboxManager = mailboxManager;
+                                 MessageFactory messageFactory,
+                                 SystemMailboxesProvider systemMailboxesProvider) {
         this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
         this.mimeMessageConverter = mimeMessageConverter;
         this.mailSpool = mailSpool;
         this.mailFactory = mailFactory;
         this.messageFactory = messageFactory;
+        this.systemMailboxesProvider = systemMailboxesProvider;
     }
 
     @Override
     public SetMessagesResponse process(SetMessagesRequest request, MailboxSession mailboxSession) {
-        Mailbox outbox;
+        Builder responseBuilder = SetMessagesResponse.builder();
+        request.getCreate()
+            .stream()
+            .forEach(create -> handleCreate(create, responseBuilder, mailboxSession));
+        return responseBuilder.build();
+    }
+
+    private void handleCreate(CreationMessageEntry create, Builder responseBuilder, MailboxSession mailboxSession) {
         try {
-            outbox = getOutbox(mailboxSession).orElseThrow(() -> new MailboxRoleNotFoundException(Role.OUTBOX));
-        } catch (MailboxException | MailboxRoleNotFoundException e) {
-            LOGGER.error("Unable to find a mailbox with role 'outbox'!");
-            throw Throwables.propagate(e);
-        }
+            validateImplementedFeature(create, mailboxSession);
+            validateArguments(create);
+            validateRights(create, mailboxSession);
+            MessageWithId created = handleOutboxMessages(create, mailboxSession);
+            responseBuilder.created(created.getCreationId(), created.getValue());
 
-        List<String> allowedSenders = ImmutableList.of(mailboxSession.getUser().getUserName());
+        } catch (MailboxUnhandledRequestException e) {
+            responseBuilder.notCreated(create.getCreationId(), 
+                    SetError.builder()
+                        .type("invalidProperties")
+                        .properties(MessageProperty.mailboxIds)
+                        .description("This mailbox can't handle this request")
```

Comment:
It should be an IllegalStateException as it must already be checked before `handleOutboxMessages` 
I just added a test for that